### PR TITLE
Use `null_resource` trigger in order to detect `ansible-playbook` command changes

### PR DESCRIPTION
**Pull request by SweetOps**
## What
* Reworked code of module to generate a trigger from the argument signature of the command

## Why
* Ansible provisioner module doesn't recognize changes when its options are changed

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+  triggers = {
+    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
+  }
+}
+
 resource "null_resource" "provisioner" {
   count = "${signum(length(var.playbook)) == 1 ? 1 : 0}"
 
   triggers {
-    default = "${sha256(file(var.playbook))}"
```
Comment (osterman) [inline main.tf]: We still want this trigger. Call this `sha256` instead of `default`

File: main.tf [hunk 2]
```diff
@@ -1,3 +1,9 @@
+resource "null_resource" "ansible" {
+  triggers = {
+    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
```
Comment (osterman) [inline main.tf]: Is it not possible to add this trigger to the `null_resource.provisioner`?

File: main.tf [hunk 3]
```diff
@@ -1,18 +1,13 @@
...
-  triggers = {
-    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
-  }
-}
-
 resource "null_resource" "provisioner" {
   count = "${signum(length(var.playbook)) == 1 ? 1 : 0}"
 
   triggers {
     default = "${sha256(file(var.playbook))}"
```
Comment (osterman) [inline main.tf]: Let's call this something other that `default`.... maybe `signature`, or `serial` or `nonce`

File: main.tf [hunk 4]
```diff
@@ -1,18 +1,13 @@
...
+    command = "ansible-playbook ${var.dry_run ? "--check --diff" : ""} ${join(" ", var.arguments)} -e ${join(" -e ", var.envs)} ${var.playbook}"
```
Comment (osterman) [inline main.tf]: Let's call this something other that `default`.... maybe `signature`, or `serial` or `nonce`

File: main.tf (hunk 3 above)
Comment (osterman) [inline main.tf]: Let's call this something other that `default`.... maybe `signature`, or `serial` or `nonce`

**Comment by SweetOps** [comment]
screenshot 2017-09-05 13 49 33

**Pull request merged**